app/database_etl/etl_main.py

```diff
@@ -254,13 +251,63 @@ def get_coords(place_name, place_type):
                 break
     return coords
 
+
 def add_latlng_to_df(place_type, place_type_name, df):
     df['coords'] = df[place_type_name].map(lambda a: get_coords(a, place_type))
     df['longitude'] = df['coords'].map(lambda a: a[0] if isinstance(a, list) else None)
     df['latitude'] = df['coords'].map(lambda a: a[1] if isinstance(a, list) else None)
     df = df.drop(columns=['coords'])
     return df
 
+
+def get_most_recent_pub_date_info(row):
+    # Get index of most recent pub date if the pub date is not None
+    try:
+        pub_dates = row['publication_date']
+        max_index = pub_dates.index(max(pub_dates))
+        row['publication_date'] = row['publication_date'][max_index]
+
+    # If pub date is None set to index to 0
+    except AttributeError:
+        max_index = 0
+
+    # If source type exists, get element at that index
+    if row['source_type']:
+        row['source_type'] = row['source_type'][max_index]
+
+    # Index whether org author exists and corresponding first author
+    is_org_author = row['organizational_author'][max_index]
+    row['organizational_author'] = is_org_author
+    row['first_author'] = row['first_author'][max_index]
+
+    # If it is not an organizational author, then get last name
+    if not is_org_author:
+        row['first_author'] = row['first_author'].strip().split()[-1]
```

Comment:
Need to make sure that .strip().split()[-1] works with an empty string. I think that it would give you index out of range.